### PR TITLE
New transforming and web-serving genegraph architecture config

### PR DESCRIPTION
**Pull request by toneillbroad**

### CODE REVIEW

**Review by tnavatar — APPROVED** [review]
This looks like the right set of choices for right now to get the feature out the door. I think it does point at some future refactoring around making different modes for Genegraph more modular, rather than requiring functionality distributed across different files in source, but I think that can be a task to consider for the future.

File: src/genegraph/server.clj [hunk 1]
```diff
@@ -19,7 +19,7 @@
      :get (fn [_] {:status 200 :body "server is live"})
      :route-name ::liveness]
     ["/ready"
-     :get (fn [_] (if (and @initialized? (stream/up-to-date?))
```
[the remark this answers is not in the document]
Comment (toneillbroad) [reply src/genegraph/server.clj:22]: Hmmm... good catch, thanks. Not sure if this is an oversite, or if there was a real reason for this. Will need to reconsider this.

File: src/genegraph/service.clj [hunk 2]
```diff
@@ -110,7 +110,8 @@
                             :request-ip-addr (get-in ctx [:request :remote-addr])
                             :servlet-request (.toString (.getRequestURL (get-in ctx [:request :servlet-request])))
                             :servlet-request-body (get-in ctx [:request :body])
-                            :reponse-status (.getStatus (get-in ctx [:request :servlet-response]))
+                            :reponse-status (some-> (get-in ctx [:request :servlet-response])
+                                                    .getStatus)
```
[the remark this answers is not in the document]
Comment (toneillbroad) [reply src/genegraph/service.clj:114]: This code was causing NullPointer exceptions and I believe this was the right fix.

File: src/genegraph/database/query.clj [hunk 3]
```diff
@@ -79,10 +79,12 @@
    (resource/construct query-string params model)))
 
 (defn get-named-graph [name]
-  (.getNamedModel db name))
+  (tx
+   (.getNamedModel db name)))
```
[the remark this answers is not in the document]
Comment (toneillbroad) [reply src/genegraph/database/query.clj:83]: Will need to check this.. I recall it is related to tests, so maybe moving it where it is needed in the tests would make more sense. Thanks.

File: src/genegraph/server.clj (hunk 1 above)
[the remark this answers is not in the document]
Comment (toneillbroad) [reply src/genegraph/server.clj:22]: Added check for update back in

File: src/genegraph/database/query.clj (hunk 3 above)
[the remark this answers is not in the document]
Comment (toneillbroad) [reply src/genegraph/database/query.clj]: Removed tx macro usage

**Pull request merged**